experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -437,22 +480,24 @@ function proj_of_image_ideal(G::LinearlyReductiveGroup, rep_mat::MatElem)
     W = image_ideal(G, rep_mat)
     mixed_ring_xy = base_ring(W[2])
     n = ncols(rep_mat)
-    m = G.group[2]
-    #use parallelised groebner bases here. This is the bottleneck!
-    return eliminate(W[1], gens(mixed_ring_xy)[(2*n)+1:(2*n)+(m^2)]), W[2]
+    return eliminate(W[1], gens(mixed_ring_xy)[(2*n)+1:end]), W[2]
 end
 
 #this function gets the generators of the null cone. they may or may not be invariant.
 #to do this we evaluate what is returned from proj_of_image_ideal at y = 0 
 #ie at gens(basering)[n+1:2*n] = [0 for i in 1:n]
 function generators(G::LinearlyReductiveGroup, X::MPolyIdeal, rep_mat::MatElem)
     n = ncols(rep_mat)
-    m = G.group[2]
+    if isdefined(G, :group)
+        m = (group_dim(G))^2
+    else
+        m = ngens(base_ring(group_ideal(G)))
+    end
     gbasis = gens(X) 
     length(gbasis) == 0 && return gbasis
     mixed_ring_xy = parent(gbasis[1])
     #evaluate at gens(mixed_ring_xy)[n+1:2*n] = 0
-    V = vcat(gens(mixed_ring_xy)[1:n], [0 for i in 1:n], gens(mixed_ring_xy)[2*n+1:2*n+m^2])
+    V = vcat(gens(mixed_ring_xy)[1:n], [0 for i in 1:n], gens(mixed_ring_xy)[2*n+1:2*n+m])
```

Comment:
Can't you also do this here, and then get rid of the definition of `m` above?
```suggestion
    V = vcat(gens(mixed_ring_xy)[1:n], [0 for i in 1:n], gens(mixed_ring_xy)[2*n+1:end])
```
also `gens` allocates, so even better:
```suggestion
    xyz = gens(mixed_ring_xy)
    V = vcat(xyz[1:n], [0 for i in 1:n], xyz[2*n+1:end])
```